R/symptome_compartment_class.R

```diff
@@ -0,0 +1,226 @@
+#' An SEIR model for compartments of different symptoms
+#'
+#' Note:
+#' 1) Sum of poulation equals to 1 at any time point.
+#' An error will be reported the sum of initial population is not equal to 1.
+#' 2) At this stage, we only consider disease-caused death cases
+#' calculating mortality.
+#'
+#' model structure:
+#'    ---------------------------(omega)---------------------------------
+#'    |                                                                 |
+#'    |                                                                 |
+#'    --> S --(lam)--> E --(gamma) ------(1-ihr)-------> I ---(nu.i)--> R <--
+#'                                 |                     |                  |
+#'                                 |                 (pdeath.i)             |
+#'                                 |                     |                  |
+#'                                 |            --> CM <--                  |
+#'                                 |            |                           |
+#'                                 |      (pdeath.hsa)                      |
+#'                                 |            |                           |
+#'                                 -- (ihr)---> HA --(nu.hsa)----------------
+
+#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Class definitions
+#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+#' @describeIn model_symptome to set up dependent classes for model_symptome
+setClass("ggplot")
```

Comment:
What do these two setClasses do?

Reply:
The S4 class model object contains a gg class object for storing a ggplot of the simulation outcome: 
`plot_output = "gg"`
The classes ggplot and gg need to be explicitly set as they don't belong to default object types.

Reply:
Ok makes sense. I guess if we change the plotting to happen outside of the simulation function, this won't be the case though?